pipes/WDL/workflows/demux_plus.wdl

```diff
@@ -13,38 +14,59 @@ workflow demux_plus {
     Array[File]? bmtaggerDbs  # .tar.gz, .tgz, .tar.bz2, .tar.lz4, .fasta, or .fasta.gz
     Array[File]? blastDbs  # .tar.gz, .tgz, .tar.bz2, .tar.lz4, .fasta, or .fasta.gz
     Array[File]? bwaDbs
+    File krakenuniq_db_tar_lz4
+    File krona_taxonomy_db_tgz
+
     scatter(raw_reads in illumina_demux.raw_reads_unaligned_bams) {
+        # de-duplicate raw reads
+        call reads.dedup_bam as dedup {
+            input:
+                in_bam = raw_reads        
+        }
+ 
+        # count spike-ins in the sample
+        #   NB: the spike-in report is created from raw reads 
+        #       that have NOT been de-duplicated
         call reports.spikein_report as spikein {
             input:
                 reads_bam = raw_reads,
                 spikein_db = spikein_db
         }
+
+        # deplete human/host genomic reads
         call taxon_filter.deplete_taxa as deplete {
             input:
-                raw_reads_unmapped_bam = raw_reads,
+                raw_reads_unmapped_bam = dedup.dedup_bam,
                 bmtaggerDbs = bmtaggerDbs,
                 blastDbs = blastDbs,
                 bwaDbs = bwaDbs
         }
+
+        # create de novo contigs from depleted reads via spaces
         call assembly.assemble as spades {
             input:
                 assembler = "spades",
                 reads_unmapped_bam = deplete.cleaned_bam,
                 trim_clip_db = trim_clip_db,
                 always_succeed = true
         }
-    }
 
-    call metagenomics.krakenuniq as krakenuniq {
-        input:
-            reads_unmapped_bam = illumina_demux.raw_reads_unaligned_bams
+        # classify de-duplicated reads to taxa via krakenuniq
+        call metagenomics.krakenuniq as krakenuniq {
```

Comment:
Oh, I don't think I'd bump this one inside the scatter. Unless maybe you want to take the opportunity to collect some data from this branch on cost efficiency changes (I've always been curious). The kraken and krakenuniq dbs have always burned about 10-15mins per job on db localization and unpacking -- which, when multiplied by the number of samples on highmem machines, adds up (which motivated the batched approach). But on principle I've always wanted to improve our staging time and move it within the scatter, but I guess I always assumed we'd never get there until we move to kraken2 (much smaller databases). That said, what's the dollars-per-demux-plus on our test/CI flowcell comparing scattered kraken vs batched kraken? If it's only a couple bucks extra, I might be fine with moving there now.

Reply:
I was curious of the cost delta, but CI builds don't show us costs since they're billed to the DNAnexus Science Team. I'll just move it back outside the scatter...